src/hasher_impl.rs

```diff
@@ -347,15 +348,21 @@ pub mod identity {
     /// # Panics
     ///
     /// Panics if the input is bigger than the maximum size.
-    #[derive(Debug, Default)]
-    pub struct IdentityHasher<S: Size> {
-        bytes: GenericArray<u8, S>,
+    #[derive(Debug)]
+    pub struct IdentityHasher<const S: usize> {
+        bytes: [u8; S],
         i: usize,
     }
 
-    impl<S: Size> StatefulHasher for IdentityHasher<S> {
-        type Size = S;
-        type Digest = IdentityDigest<Self::Size>;
+    impl<const S: usize> Default for IdentityHasher<S> {
+        fn default() -> Self {
+            Self {i: 0, bytes: [0u8; S]}
+        }
+    }
```

Comment:
Why doesn't derive work?

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d19c29c843a9a840ab6f40f831c53150

We might be able to implement Default for [u8; S] but I haven't come across the right way to do it though.